intermine/query_manager.py

```diff
@@ -0,0 +1,235 @@
+        return ", ".join(result)
+
+
+def delete_query(name):
+    """
+    A function that deletes a given query
+    ================================================
+    example:
+
+        >>>from intermine import query_manager as qm
+        >>>qm.delete_query('queryName')
+        <deletes the query whose name is 'queryName' from user's account>
+
+    """
+    # mock z for testing
+    if mine == 'mock':
+        z = {'queries': {'query1': 1, 'query2': 2}}
+    else:
+        # source of the initial request
+        x = "http://registry.intermine.org/service/instances/" + mine
+        r = requests.get(x)
+        dict = json.loads(r.text)
+        # source of the next request
+        y = dict["instance"]["url"] + "/service/user/queries?token=" + token
+        r = requests.get(y)
+        # dictionary form of data
+        z = json.loads(r.text)
+    # checks if query name exists
+    count = 0
+    for key in z['queries'].keys():
+        if key == name:
+            count = count + 1
+    if count == 0:
+        return "No such query available"
+
+    else:
+        # returns just a message for tests
+        if mine == 'mock':
+            return name + " is deleted"
+        else:
+            link = dict["instance"]["url"] + "/service/user/queries/" + name +\
+             "?token=" + token
+            requests.delete(link)
+            return name + " is deleted"
+
+
+def post_query(xml):
+    """
+    A function to post a query(in the form of string containing xml)
+    to a user account
+    ================================================
+    example:
+
+        >>>qm.post_query('<query name="" model="genomic" view="Gene.length\
+            Gene.symbol" longDescription="" sortOrder="Gene.length asc">\
+            </query>')
+    Note that the name should be defined first
+    """
+    # xml parsing
+    root = etree.fromstring(xml)
+    # mock raw for testing
+    if mine == 'mock':
+        raw = {'queries': {'query1': 1, 'query2': 2}}
+    else:
+        # source of the initial request
+        x = "http://registry.intermine.org/service/instances/" + mine
+        r = requests.get(x)
+        dict = json.loads(r.text)
+        link = dict["instance"]["url"] + "/service/user/queries?token=" + token
+        r = requests.get(link)
+        raw = json.loads(r.text)
+    count = 0
+    for key in raw['queries'].keys():
+        if key == root.attrib['name']:
+            count = count + 1
+            print("The query name exists")
+            try:
+                resp = input("Do you want to replace the old query? [y/n]")
+            except NameError:
+                resp = raw_input("Please enter response again [y/n]")
+            if resp == 'y':
+                count = 0
+
+    if count == 0:
+        # mock raw for testing
+        if mine == 'mock':
+            raw = {'queries': {'query1': 1, 'query2': 2, 'query3': 3}}
+        else:
+            link = dict["instance"]["url"] + "/service/user/queries?xml=" + \
```

Comment:
this parameter `xml` needs to be changed because it now accepts JSON queries. See https://github.com/intermine/intermine/pull/1815

What can we do? 

Many mines will have the `xml` parameter a long time. But it doesn't make any sense to have a parameter called XML that accepts JSON.

Do we have any ideas for solutions?